### ISSUE TITLE
Multiplicity constraint error when using HadoopFactory

### ISSUE THREAD

**Issue by rjbriody** [thinkaurelius/titan#807]
Here is the relevant discussion: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/aureliusgraphs/vLHe43s6XlA/sKFW2ji6DlcJ

Step 1 - add schema

```
graph = <get a cassandra connected graph>
mgmt = graph.getManagementSystem()
mgmt.makePropertyKey("email").dataType(String).cardinality(com.thinkaurelius.titan.core.Cardinality.LIST).make()
mgmt.commit()
graph.commit()
```

Step 2 - add vertex with multi-property

```
v = graph.addVertex(null)
v.addProperty("email", "one")
v.addProperty("email", "two")
graph.commit()
```

Step 3 - boom!

```
g = HadoopFactory.open('../conf/hadoop/titan-cassandra-input.properties')
g._
```

Results in following exception:

```
Error: java.lang.RuntimeException: java.lang.IllegalArgumentException: A relation already exists which violates the multiplicity constraint: MANY2ONE on type freebase_name at com.thinkaurelius.titan.hadoop.formats.util.TitanHadoopGraph.readHadoopVertex(TitanHadoopGraph.java:102) at com.thinkaurelius.titan.hadoop.formats.cassandra.TitanCassandraHadoopGraph.readHadoopVertex(TitanCassandraHadoopGraph.java:30) at com.thinkaurelius.titan.hadoop.formats.cassandra.TitanCassandraRecordReader.nextKeyValue(TitanCassandraRecordReader.java:49) at org.apache.hadoop.mapred.MapTask$NewTrackingRecordReader.nextKeyValue(MapTask.java:533) at org.apache.hadoop.mapreduce.task.MapContextImpl.nextKeyValue(MapContextImpl.java:80) at org.apache.hadoop.mapreduce.lib.map.WrappedMapper$Context.nextKeyValue(WrappedMapper.java:91) at org.apache.hadoop.mapreduce.lib.chain.Chain$ChainRecordReader.nextKeyValue(Chain.java:156) at org.apache.hadoop.mapreduce.lib.chain.ChainMapContextImpl.nextKeyValue(ChainMapContextImpl.java:78) at org.apache.hadoop.mapreduce.lib.map.WrappedMapper$Context.nextKeyValue(WrappedMapper.java:91) at org.apache.hadoop.mapreduce.Mapper.run(Mapper.java:144) at org.apache.hadoop.mapreduce.lib.chain.Chain.runMapper(Chain.java:389) at org.apache.hadoop.mapreduce.lib.chain.ChainMapper.run(ChainMapper.java:149) at org.apache.hadoop.mapred.MapTask.runNewMapper(MapTask.java:764) at org.apache.hadoop.mapred.MapTask.run(MapTask.java:340) at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:167) at java.security.AccessController.doPrivileged(Native Method) at javax.security.auth.Subject.doAs(Subject.java:415) at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1554) at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:162) Caused by: java.lang.IllegalArgumentException: A relation already exists which violates the multiplicity constraint: MANY2ONE on type freebase_name at com.thinkaurelius.titan.hadoop.FaunusElement.ensureUniqueness(FaunusElement.java:248) at com.thinkaurelius.titan.hadoop.FaunusElement.addRelation(FaunusElement.java:205) at com.thinkaurelius.titan.hadoop.FaunusVertex.addProperty(FaunusVertex.java:122) at com.thinkaurelius.titan.hadoop.formats.util.TitanHadoopGraph.readHadoopVertex(TitanHadoopGraph.java:71) ... 18 more 
```

**Comment by philmes** [comment]
If anyone else is being afflicted by this problem, I have temporarily got around it by removing the multiplicity check in `com.thinkaurelius.titan.hadoop.FaunusElement.addRelation`. I figure this is safe enough for my use case, which is currently just read-only analytics. Probably more risky if you're writing data into Titan/Cassandra.

**Comment by mbroecheler** [comment]
@dalaro This is probably due to schema information getting lost between iterations. We should remove any schema checking and only enforce it when writing data back into the graph.

**Comment by dalaro** [comment]
This is reproducible on 0.5.1-hadoop2, but it does not occur on titan05 as of this morning.  This might have been fixed as part of #786.

Since this is working currently on titan05, I added a test that would break if this regresses in 856a84ee55aa106d29db3d6ade540b7ca627dd57.
